webpack.config.js

```diff
@@ -0,0 +1,97 @@
+"use strict";
+const webpack = require("webpack");
+const { BundleAnalyzerPlugin } = require("webpack-bundle-analyzer");
+const { merge } = require("webpack-merge");
+
+const bnJsReplaces = [
+  "tiny-secp256k1",
+  "asn1.js",
+  "create-ecdh",
+  "miller-rabin",
+  "public-encrypt",
+  "elliptic",
+];
+
+function getDefaultConfiguration() {
+  return {
+    cache: true,
+    performance: {
+      hints: false,
+      maxEntrypointSize: 512000,
+      maxAssetSize: 512000,
+    },
+    stats: "errors-only",
+    devtool: "source-map",
+    plugins: [
+      new webpack.ProvidePlugin({ process: "process/browser" }),
+      new webpack.ProvidePlugin({ Buffer: ["buffer", "Buffer"] }),
+      // this is a bit of a hack to prevent 'bn.js' from being installed 6 times
+      // TODO: any package that is updated to use bn.js 5.x needs to be removed from `bnJsReplaces` above
+      // https://github.com/webpack/webpack/issues/5593#issuecomment-390356276
+      new webpack.NormalModuleReplacementPlugin(/^bn.js$/, (resource) => {
```

Comment:
Maybe not for this PR - but can we resolve this TODO now?

Reply:
This logic will be removed by #2273 but I added it to this PR so that neither is holding the other back.